extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -169,32 +171,35 @@ export class AmpGeo extends AMP.BaseElement {
    * @param {Document} doc
    */
   findCountry_(doc) {
-    // First see if we've been pre-rendered with a country, if so set it
+    // Flag to see if we've been pre-rendered with a country
     const preRenderMatch = doc.body.className.match(PRE_RENDER_REGEX);
+    // Trim the spaces off the patched country
+    const trimmedCountry = COUNTRY.trim();
 
-    if (preRenderMatch &&
-        !Services.urlForDoc(this.element).isProxyOrigin(doc.location)) {
-      this.mode_ = mode.GEO_PRERENDER;
-      this.country_ = preRenderMatch[1];
-    } else {
-      this.mode_ = mode.GEO_HOT_PATCH;
-      this.country_ = COUNTRY.trim();
-      // If we got a country code it will be 2 characters
-      // If the lengths is 0 the country is unknown
-      // If the length is > 2 we didn't get patched
-      // (probably local dev) so we treat it as unknown.
-      if (this.country_.length !== 2) {
-        this.country_ = 'unknown';
-      }
-    }
+    // default country is 'unknown' which is also the zero length case
 
-    // Are we in debug override?
-    // match to \w characters only to prevent xss vector
     if (getMode(this.win).geoOverride &&
       (isCanary(this.win) || getMode(this.win).localDev) &&
       /^\w+$/.test(getMode(this.win).geoOverride)) {
+      // debug override case, only works in canary or localdev
+      // match to \w characters only to prevent xss vector
       this.mode_ = mode.GEO_OVERRIDE;
       this.country_ = getMode(this.win).geoOverride.toLowerCase();
+    } else if (preRenderMatch &&
+        !Services.urlForDoc(this.element).isProxyOrigin(doc.location)) {
+      // pre-rendered by a cache case
+      this.mode_ = mode.GEO_PRERENDER;
+      this.country_ = preRenderMatch[1];
+    } else if (trimmedCountry.length == 2) {
+      // We have a valid 2 letter ISO country
+      this.mode_ = mode.GEO_HOT_PATCH;
+      this.country_ = trimmedCountry;
+    } else if (trimmedCountry.length > 2 && !getMode(this.win).localDev) {
+      // We were not patched, if we're not in dev this is an error
+      // and we leave the country at the default 'unknown'
+      this.error_ = true;
+      dev().error(TAG,
+          'GEONOTPATCHED: amp-geo served unpatched, ISO country not set');
```

Comment:
Ditto, there should be an `else` (covering `trimmedCountry.length > 2 && localDev`) that sets `country_` to `'unknown'`.

Reply:
Set to `unknown` in the constructor